### PR TITLE
new endpoint to retrieve template

### PR DESCRIPTION
**Pull request by marijnritense**

### CODE REVIEW

File: document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java [hunk 1]
```diff
@@ -61,6 +61,22 @@ public ResponseEntity<Page<? extends DocumentDefinition>> getDocumentDefinitions
         return ok(documentDefinitionService.findAll(fixPageable(pageable)));
     }
 
+    @Override
+    public ResponseEntity<String> getDocumentDefinitionTemplate() {
```
Comment (theo-ritense) [inline document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java]: I guess the story was not clear. The `$id` and `title` should be provided by the FE and replaced in the template. I believe that also means this should be a POST?

Comment (marijnritense) [reply document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java]: Fixed

File: document/src/main/java/com/ritense/document/domain/impl/template/DocumentDefinitionTemplateRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
```
Comment (theo-ritense) [inline document/src/main/java/com/ritense/document/domain/impl/template/DocumentDefinitionTemplateRequestDto.java]:
```suggestion
 * Copyright 2015-2024 Ritense BV, the Netherlands.
```

File: document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java [hunk 3]
```diff
@@ -61,6 +62,22 @@ public ResponseEntity<Page<? extends DocumentDefinition>> getDocumentDefinitions
         return ok(documentDefinitionService.findAll(fixPageable(pageable)));
     }
 
+    @Override
+    public ResponseEntity<String> getDocumentDefinitionTemplate(DocumentDefinitionTemplateRequestDto requestDto) {
+        return ok(
+            """
+                {
+                    "$id": "%s.schema",
```
Comment (theo-ritense) [inline document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java:70]: Shouldn't we sanatize the input? I could break this JSON by sending `xyz"`. Best would be to use Jackson to set/add a TextNode I think.

File: document/src/main/java/com/ritense/document/domain/impl/template/DocumentDefinitionTemplateRequestDto.java (hunk 2 above)
Comment (marijnritense) [reply document/src/main/java/com/ritense/document/domain/impl/template/DocumentDefinitionTemplateRequestDto.java]: Updated

File: document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java (hunk 3 above)
Comment (marijnritense) [reply document/src/main/java/com/ritense/document/web/rest/impl/JsonSchemaDocumentDefinitionResource.java:70]: Added sanitization

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**